### PR TITLE
Hibernate search 5.9.1 and sortable

### PR DESCRIPTION
**Pull request by olliefreeman**
* Updates to 5.9.1
* Adds sortable fields
* Adds simple query strings
* Updates/refactors code
* Removes filter as this has now been deprecated
* Updated readme
* Updated version to 2.3.0-SNAPSHOT

### CODE REVIEW

**Comment by lgrignon** [comment]
Thank you for your contribution! It seems that you did a pretty amazing job with this upgrade / refactoring.

How much did you test? Is it possible that you try to use your plugin on this showcase project and PR?: https://github.com/lgrignon/grails3-quick-start ? That would be great. I will also test it on another project I have to guarantee maximum testing.

Thanks again

File: README.md [hunk 1]
```diff
@@ -103,14 +115,14 @@ class MyDomainClass {
 
     static hasMany = [categories: Category, items: Item]
 
-    static search = {
+    static luceneIndexing = {
```
Comment (lgrignon) [inline README.md]: I think the former name was better, since Hibernate Search could rely on other engine than lucene (such as ElasticSearch) and we don't want all users to rename their closures :) Could you switch back to "search" please?

File: README.md [hunk 2]
```diff
@@ -4,39 +4,51 @@ This plugin aims to integrate Hibernate Search features to Grails in very few st
...
 
-
-
 ## Getting started
 
 If you don't want to start from the [template project](#examples), you could start a fresh project:
 
-Add the following repository
-```
-  repositories { maven { url "http://dl.bintray.com/lgrignon/plugins" } }
```
Comment (lgrignon) [inline README.md]: Is this not required anymore? I published the plugin on the official repository so I guess this is ok

**Comment by olliefreeman** [comment]
We have a (private) project which extensively uses the hibernate search functionality and i'm in the process of running its tests through now. However I have manually tested the changes along with updating our project to use the simple query strings and done some manual testing.

If i had more time i would write some tests for the plugin however unfortunately I have run out of time :(

I will try and add/update to the quick start repo.

File: README.md [hunk 3]
```diff
@@ -347,25 +345,67 @@ class SomeController {
...
        setFetchMode("authors", FetchMode.JOIN)
     }
-    
+
     fuzzy "description", "mi search"
   }
 ```
 
-#### Sorting the results
+### Performing SimpleQueryString searches
```
Comment (lgrignon) [inline README.md]: Very nice! :)

**Comment by lgrignon** [comment]
Awesome, thanks.

File: README.md (hunk 2 above)
Comment (olliefreeman) [reply README.md]: yup, the plugin is available through the standard grails repo

File: README.md (hunk 1 above)
Comment (lgrignon) [reply README.md]: Oh, I see. I think the point was actually to have the same name to make it simple to use and understand, but if you really think it could collide in some way, I would then vote for hibernateSearchIndexing.

Honestly, I gave up on the Groovy tooling (mostly because I use Eclipse / cannot afford Intellij :)), I just try to use @GrailsCompileStatic as much as I can, but since we are using a dynamically added methods here, I don't think we can do anything. I am not aware of such dynamic descriptors (such as grails plugins describing which methods they could add) but I doubt it exists, and even if, not sure that tooling would use it.

Anyway, thanks for renaming, I will merge after it, and try on other project

Comment (olliefreeman) [reply README.md]: The collision was most noticeable when running application in live-reload, as then you'd get errors saying "cannot apply list()" to xxxx. That error is actually solved by the updates to the plugin class and the use of onChange, so the actual collision is now gone.

So i'll name it back to `search` and we've got the section in the readme `IDE Integration` to help tooling

**Comment by olliefreeman** [comment]
pushed updates to resolve comments and missing README section

**Comment by olliefreeman** [comment]
also all tests on our project have passed

**Comment by lgrignon** [comment]
Released to:
https://bintray.com/lgrignon/plugins/hibernate-search/2.3.0

Showcase updated:
https://github.com/lgrignon/grails3-quick-start/commit/1c6671e1f0f5ee4b8ed2b34b8d260485fd0133dd

Thanks again for your contribution!
